### ISSUE TITLE
Mail fetching fails if ticket is to be denied

### ISSUE THREAD

**Issue by Lineflyer** [osTicket/osTicket#6441]
### Prerequisites

* [x] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [x] Do you have any errors in the PHP error log, or javascript console?
* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

Today I realized, that osticket v1.17.2 stopped pulling new mails via IMAP.
The system is configured with a remote cron triggered each 5 minutes.

Seemingly the reason was, that within the mail to be pulled there was a (spam) mail being on the ban list.
This somehow broke the complete mail fetching.

### Steps to Reproduce

1. Have some mails in your IMAP mailbox
2. One of them is received from a mail address on the system ban list
3. Trigger a remote cron to initiate pulling the mails.

**Expected behavior:**
Mails should be fetched and converted to tickets. 
Mails from sender on the ban list should be denied during that process.

As long as there is no banned mail the remote cron runs normal and reliable.

**Actual behavior:**
The system stopped fetching any mails and logs API errors.

System log shows (repeatedly for each cron execution):
![image](https://user-images.githubusercontent.com/949669/213572750-027f5653-335a-4bd9-8c60-651930b6bbaf.png)

This continues for days without any ticket being created.
The existing mails (including the spam) remain unprocessed on the mail server.

Also autocron does not work. I got it going again only after manually deleting the problematic mail from the inbox on the server.

Details (identical on each logging instance with same time stamp):
![image](https://user-images.githubusercontent.com/949669/213573247-56cc5db3-3d57-4b6f-aaa5-f930f1ce0d03.png)
![image](https://user-images.githubusercontent.com/949669/213573387-e0e3762f-00d6-48fc-96f1-b88fe890288d.png)

### Versions

![image](https://user-images.githubusercontent.com/949669/213574314-2c6f8ce9-0715-4f2e-ad90-f7ba3c2eec12.png)

rcron.php (URL and key removed):
```
#!/usr/bin/php -q
<?php
/*********************************************************************
    rcron.php

    PHP script used for remote cron calls.

    Peter Rotich <peter@osticket.com>
    Copyright (c)  2006-2013 osTicket
    http://www.osticket.com

    Released under the GNU General Public License WITHOUT ANY WARRANTY.
    See LICENSE.TXT for details.

    vim: expandtab sw=4 ts=4 sts=4:
**********************************************************************/

# Configuration: Enter the url and key. That is it.
#  url => URL to api/task/cron e.g http://yourdomain.com/support/api/tasks/cron
#  key => API's Key (see admin panel on how to generate a key)
#

$config = array(
        'url'=>'https://MYURL/api/tasks/cron',
        'key'=>'MYKEY'
        );

#pre-checks
function_exists('curl_version') or die('CURL support required');

#set timeout
set_time_limit(30);

#curl post
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $config['url']);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, '');
curl_setopt($ch, CURLOPT_USERAGENT, 'osTicket API Client v1.7');
curl_setopt($ch, CURLOPT_HEADER, TRUE);
curl_setopt($ch, CURLOPT_HTTPHEADER, array( 'Expect:', 'X-API-Key: '.$config['key']));
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, FALSE);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
$result=curl_exec($ch);
curl_close($ch);

if(preg_match('/HTTP\/.* ([0-9]+) .*/', $result, $status) && $status[1] == 200)
    exit(0);

echo $result;
exit(1);
?>
```

**Comment by JediKev** [comment]
@Lineflyer 

You'll need these:
- https://github.com/osTicket/osTicket/pull/6381
- https://github.com/osTicket/osTicket/pull/6399
- https://github.com/osTicket/osTicket/pull/6409

Cheers.

**Comment by Lineflyer** [comment]
Great, will take a look into it.
Any plans on a release upcoming on short term or shall I apply the mentioned PRs as patches manually?

**Comment by JediKev** [comment]
@Lineflyer 

Next release should be within the next couple of weeks.

Cheers.

**Comment by Lineflyer** [comment]
@JediKev 
Can you confirm, that this problem is fixed within the normal releases meanwhile?
Meaning: Should this issue be fixed when upgrading to v1.17.5 respectively v1.18.1?

**Comment by JediKev** [comment]
@Lineflyer 

Yes, that was addressed in subsequent releases.

Cheers.
